xdsl/ir.py

```diff
@@ -231,13 +235,12 @@ def owner(self) -> Block:
         return self.block
 
     def __repr__(self) -> str:
-        if isinstance(self.block, Block):
-            block_repr = f"Block(num_arguments={len(self.block.args)}, " + \
-                         f"num_blocks={len(self.block.ops)} ops)"
-        else:
-            block_repr = repr(self.block)
-        return f"OpResult(typ={repr(self.typ)}, num_uses={repr(len(self.uses))}" + \
```

Comment:
IMO, to make this less controversial, I would first start with a small PR changin `OpResult` to `BlocKArgument` here. This is clearly a bug in the naming. Also the whole if-else thing doesn't make sene, IMO.